framework/src/base/FEProblemBase.C

```diff
@@ -160,7 +160,8 @@ FEProblemBase::FEProblemBase(const InputParameters & parameters) :
     _force_restart(getParam<bool>("force_restart")),
     _fail_next_linear_convergence_check(false),
     _currently_computing_jacobian(false),
-    _started_initial_setup(false)
+    _started_initial_setup(false),
+    _cache_mesh_changed(false)
```

Comment:
See - this is what I mean: `cache_mesh_changed` sounds like it's going "cache whether or not the mesh changed"